### PR TITLE
Reduce Deadlock Probability

### PR DESCRIPTION
**Pull request by franchuterivera**
Use the logger port instead of the logger for the TAE execution.

Add show_models() for debug purposes 

Remove tensorboard output because killing a run in the process of writing to disk halts the complete search process and python does not handle the recovery nice. This is something we should look fixing in pynisher.

Minor fixes like empty cuda in when not needed.

### CODE REVIEW

File: autoPyTorch/pipeline/components/setup/traditional_ml/classifier_models/classifiers.py [hunk 1]
```diff
@@ -106,6 +106,13 @@ def predict(self, X_test: np.ndarray, predict_proba: bool = False) -> np.ndarray
         y_pred = self.model.predict(X_test)
         return y_pred
 
+    @staticmethod
+    def get_properties(dataset_properties: Optional[Dict[str, Any]] = None) -> Dict[str, str]:
+        return {
+            'shortname': 'LGBMClassifier',
+            'name': 'LGBMClassifier',
```
Comment (ravinkohli) [inline autoPyTorch/pipeline/components/setup/traditional_ml/classifier_models/classifiers.py:113]: could we have the full name here?

File: autoPyTorch/pipeline/components/setup/traditional_ml/classifier_models/classifiers.py [hunk 2]
```diff
@@ -226,6 +240,13 @@ def predict(self, X_test: np.ndarray, predict_proba: bool = False) -> np.ndarray
         y_pred = self.model.predict(X_test)
         return y_pred
 
+    @staticmethod
+    def get_properties(dataset_properties: Optional[Dict[str, Any]] = None) -> Dict[str, str]:
+        return {
+            'shortname': 'RandomForestClassifier',
```
Comment (ravinkohli) [inline autoPyTorch/pipeline/components/setup/traditional_ml/classifier_models/classifiers.py:246]: could we make it RFClassifier

File: autoPyTorch/pipeline/components/setup/traditional_ml/classifier_models/classifiers.py [hunk 3]
```diff
@@ -338,6 +366,13 @@ def predict(self, X_test: np.ndarray, predict_proba: bool = False) -> np.ndarray
         y_pred = self.model.predict(X_test)
         return y_pred
 
+    @staticmethod
+    def get_properties(dataset_properties: Optional[Dict[str, Any]] = None) -> Dict[str, str]:
+        return {
+            'shortname': 'KNeighborsClassifier',
```
Comment (ravinkohli) [inline autoPyTorch/pipeline/components/setup/traditional_ml/classifier_models/classifiers.py:372]: can this be KNNClassifier

File: autoPyTorch/pipeline/components/setup/traditional_ml/classifier_models/classifiers.py [hunk 4]
```diff
@@ -338,6 +366,13 @@ def predict(self, X_test: np.ndarray, predict_proba: bool = False) -> np.ndarray
...
+            'name': 'KNeighborsClassifier',
```
Comment (ravinkohli) [inline autoPyTorch/pipeline/components/setup/traditional_ml/classifier_models/classifiers.py:373]: can this be K Nearest Neighbours Classifier

File: autoPyTorch/pipeline/components/setup/traditional_ml/classifier_models/classifiers.py [hunk 5]
```diff
@@ -384,3 +419,10 @@ def predict(self, X_test: np.ndarray, predict_proba: bool = False) -> np.ndarray
             return self.model.predict_proba(X_test)
         y_pred = self.model.predict(X_test)
         return y_pred
+
+    @staticmethod
+    def get_properties(dataset_properties: Optional[Dict[str, Any]] = None) -> Dict[str, str]:
+        return {
+            'shortname': 'SVC',
+            'name': 'SVC',
```
Comment (ravinkohli) [inline autoPyTorch/pipeline/components/setup/traditional_ml/classifier_models/classifiers.py:427]: Can this be Support Vector Classification

File: autoPyTorch/pipeline/tabular_classification.py [hunk 6]
```diff
@@ -225,6 +227,39 @@ def _get_pipeline_steps(self, dataset_properties: Optional[Dict[str, Any]],
...
+
+        It should be a representation that follows:
+        [{'PreProcessing': <>, 'Estimator': <>}]
+
+        Returns:
+            Dict: contains the pipeline representation in a short format
+        """
+        preprocessing = []
+        estimator = []
+        skip_steps = ['data_loader', 'trainer', 'lr_scheduler', 'optimizer', 'network_init',
```
Comment (ravinkohli) [inline autoPyTorch/pipeline/tabular_classification.py:243]: Maybe we can have a verbose option where we include- trainer, lr_scheduler, optimizer and network_init?

File: autoPyTorch/pipeline/tabular_regression.py [hunk 7]
```diff
@@ -172,6 +173,39 @@ def _get_pipeline_steps(self, dataset_properties: Optional[Dict[str, Any]]) -> L
...
+
+        It should be a representation that follows:
+        [{'PreProcessing': <>, 'Estimator': <>}]
+
+        Returns:
+            Dict: contains the pipeline representation in a short format
+        """
+        preprocessing = []
+        estimator = []
+        skip_steps = ['data_loader', 'trainer', 'lr_scheduler', 'optimizer', 'network_init',
```
Comment (ravinkohli) [inline autoPyTorch/pipeline/tabular_regression.py:189]: Same here

**Review by ravinkohli — COMMENTED** [review]
The PR looks great, I have just added minor comments that will give a bit more information to the user.

File: autoPyTorch/pipeline/tabular_classification.py (hunk 6 above)
Comment (franchuterivera) [reply autoPyTorch/pipeline/tabular_classification.py:243]: I was coding this and seeing the outcome, and it just doesn't add much information to 'what the best estimator is' because it's part of the construction of the model, but not part of the model itself.

I have a better proposal -- I would like to create a command like [this](https://github.com/EpistasisLab/tpot/blob/6448bdb71ba08b4a0447c640d2f05a05e1affc21/tpot/export_utils.py#L56).

Tpot is able to print the python code on how to train the model that you did. So if your goal were to see what happened (and for debug purposes) it would be great to produce a file that contains PyTorch commands with not only the scheduler but also the config (like it is nice to know we know more that the fact that we picked `adam` optimizers). If this is better, then I would like to disentangle this export_pipeline command from this PR and create an issue for this.
What do you think?

Comment (ravinkohli) [reply autoPyTorch/pipeline/tabular_classification.py:243]: yes this would be ideal to use. Could you add an issue so we don't forget?
